test/IDE/structure.swift

```diff
@@ -217,12 +217,21 @@ class ReturnType {
   // CHECK: }</ifunc>
   
   func foo2<T>() -> T {}
-  // CHECK:  <ifunc>func <name>foo2<T>()</name> -> <type>T</type> {}</ifunc>
+  // CHECK:  <ifunc>func <name>foo2<<generic-param><name>T</name></generic-param>>()</name> -> <type>T</type> {}</ifunc>
   
   func foo3() -> () -> Int {}
   // CHECK:  <ifunc>func <name>foo3()</name> -> <type>() -> Int</type> {}</ifunc>
 }
 
+// CHECK: <struct>struct <name>Generic</name><<generic-param><name>T</name>: <inherited><elem-typeref>Comparable</elem-typeref></inherited></generic-param>, <generic-param><name>X</name></generic-param>> {
+// CHECK:   <subscript><name>subscript<<generic-param><name>U</name></generic-param>>(<param>generic: <type>U</type></param>)</name> -> <type>Int</type> { return 0 }</subscript>
+// CHECK:   <typealias>typealias <name>Foo<generic-param><name>Y</name></generic-param></name> = Bar<Y></typealias>
```

Comment:
unfortunately, there's something else preventing `source.lang.swift.decl.generic_type_param` to be inside a `typealias` substructure. 

I think maybe there's something doing this since `typealias` didn't have a `key.substructure` until now.

Reply:
ASTWalker's `shouldWalkIntoGenericParams` is [documented](https://github.com/apple/swift/blob/01dff6d50989d9d8c7b0c7e8b606d992a943f40d/include/swift/AST/ASTWalker.h#L202) as applying only to nominal + function decls -- could try extending to typealias [here](https://github.com/apple/swift/blob/01dff6d50989d9d8c7b0c7e8b606d992a943f40d/lib/AST/ASTWalker.cpp#L193) to go like `visitNominalTypeDecl` below.

Reply:
@johnfairh That's for the tip! I've done that in 	e57bcc2